### PR TITLE
Update UserModelGenerator for CI 4.3.0

### PR DESCRIPTION
**Pull request by kenjis**
See #596

- update UserModelGenerator::run()
- update tests

### CODE REVIEW

**Comment by kenjis** [comment]
Some checks failed, but they will be fixed in the next PR #597

File: src/Commands/Generators/UserModelGenerator.php [hunk 1]
```diff
@@ -68,12 +68,14 @@ public function run(array $params): void
...
             CLI::error('Cannot use `ShieldUserModel` as class name as this conflicts with the parent class.', 'light_gray', 'red');
 
-            return; // @TODO when CI4 is at v4.3+, change this to `return 1;` to signify failing exit
+            return 1;
         }
 
         $params[0] = $class;
 
-        $this->execute($params);
+        $this->generateClass($params);
```
[the remark this answers is not in the document]
Comment (kenjis) [reply src/Commands/Generators/UserModelGenerator.php:76]: Good point! I will fix it.

File: tests/Commands/UserModelGeneratorTest.php [hunk 2]
```diff
@@ -18,10 +19,17 @@ protected function setUp(): void
     {
         parent::setUp();
 
-        CITestStreamFilter::$buffer = '';
+        // @phpstan-ignore-next-line
+        if (version_compare(CodeIgniter::CI_VERSION, '4.3.0', '>=')) {
```
Comment (paulbalandan) [inline tests/Commands/UserModelGeneratorTest.php:23]: Why do you need this if statement? Why not go with 4.3 and change related lines?

File: src/Commands/Generators/UserModelGenerator.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (kenjis) [reply src/Commands/Generators/UserModelGenerator.php]: Fixed.

File: tests/Commands/UserModelGeneratorTest.php (hunk 2 above)
Comment (kenjis) [reply tests/Commands/UserModelGeneratorTest.php:23]: 4.3.0 has many breaking changes. I think dropping 4.2 now is too early.

Comment (paulbalandan) [reply tests/Commands/UserModelGeneratorTest.php:23]: Okay. I think that's the reason for your PR testing against lowest dependencies, right?

Comment (kenjis) [reply tests/Commands/UserModelGeneratorTest.php:23]: Yes!

**Review by paulbalandan — APPROVED** [review]

File: tests/Commands/UserModelGeneratorTest.php [hunk 3]
```diff
@@ -18,10 +19,17 @@ protected function setUp(): void
     {
         parent::setUp();
 
-        CITestStreamFilter::$buffer = '';
+        // @phpstan-ignore-next-line
```
Comment (paulbalandan) [inline tests/Commands/UserModelGeneratorTest.php:22]: This can be added to phpstan's dynamic constants.

Comment (kenjis) [reply tests/Commands/UserModelGeneratorTest.php]: Thanks. It is better! Done.

**Review by datamweb — APPROVED** [review]
Thanks!

**Pull request merged**